### PR TITLE
SDKS-3098 Support Android 15

### PR DESCRIPTION
**Pull request by witrisna**
# JIRA Ticket

[SDKS-3098](https://bugster.forgerock.org/jira/browse/SDKS-3098)

# Description

- Update CompileSDK to API 35
- Upgrade all libraries to the latest version (except firebase-messaging)
- Test and resolve compile issue after upgrade

### CODE REVIEW

File: forgerock-auth/src/main/java/org/forgerock/android/auth/webauthn/WebAuthnRegistration.kt [hunk 1]
```diff
@@ -159,13 +159,15 @@ open class WebAuthnRegistration() : WebAuthn() {
...
         if (options.authenticatorSelection?.requireResidentKey == true &&
             options.authenticatorSelection?.residentKeyRequirement == ResidentKeyRequirement.RESIDENT_KEY_DISCOURAGED) {
-            val source = PublicKeyCredentialSource.builder()
-                .id(publicKeyCredential.rawId)
-                .rpid(options.rp.id)
-                .userHandle(Base64.decode(options.user.id, Base64.URL_SAFE or Base64.NO_WRAP))
-                .otherUI(options.user.displayName).build()
-            persist(context, source)
-        }
+            publicKeyCredential.rawId?.let {
```
Comment (rodrigoareis) [inline forgerock-auth/src/main/java/org/forgerock/android/auth/webauthn/WebAuthnRegistration.kt:162]: Is this an intentional change?

File: forgerock-auth/src/main/java/org/forgerock/android/auth/webauthn/WebAuthnRegistration.kt
Comment (rodrigoareis) [inline forgerock-auth/src/main/java/org/forgerock/android/auth/webauthn/WebAuthnRegistration.kt:1]: Update copyright

File: forgerock-auth/src/test/java/org/forgerock/android/auth/callback/AppIntegrityCallbackTest.kt
Comment (rodrigoareis) [inline forgerock-auth/src/test/java/org/forgerock/android/auth/callback/AppIntegrityCallbackTest.kt:1]: Update copyright

File: samples/app/src/main/java/com/example/app/callback/DeviceSigningVerifierCallback.kt
Comment (rodrigoareis) [inline samples/app/src/main/java/com/example/app/callback/DeviceSigningVerifierCallback.kt:1]: Update copyright

File: samples/quickstart/src/main/AndroidManifest.xml
Comment (rodrigoareis) [inline samples/quickstart/src/main/AndroidManifest.xml:1]: Update copyright

**Review by rodrigoareis — CHANGES_REQUESTED** [review]
Minor comments to address

File: forgerock-auth/src/main/java/org/forgerock/android/auth/webauthn/WebAuthnRegistration.kt (hunk 1 above)
Comment (jeyanthanperiyasamy) [reply forgerock-auth/src/main/java/org/forgerock/android/auth/webauthn/WebAuthnRegistration.kt:162]: +1

File: buildSrc/src/main/kotlin/AndroidBuildGradlePlugin.kt [hunk 2]
```diff
@@ -14,7 +14,7 @@ class AndroidBuildGradlePlugin : Plugin<Project> {
 
     override fun apply(project: Project) {
         project.android().apply {
-            compileSdk = 34;
+            compileSdk = 35;
             defaultConfig {
                 minSdk = 23
```
Comment (jeyanthanperiyasamy) [inline buildSrc/src/main/kotlin/AndroidBuildGradlePlugin.kt:19]: should we need to support 23 ? i heard the Android 15 lower than 24 can't be installed.

**Review by jeyanthanperiyasamy — COMMENTED** [review]
looks good , a minor comment

**Review by jeyanthanperiyasamy — APPROVED** [review]

**Review by spetrov — DISMISSED** [review]

File: forgerock-auth/src/main/java/org/forgerock/android/auth/webauthn/WebAuthnRegistration.kt (hunk 1 above)
Comment (witrisna) [reply forgerock-auth/src/main/java/org/forgerock/android/auth/webauthn/WebAuthnRegistration.kt:162]: Yes, it is intended, after upgrade to the new fido library, the getRawId  can return null.
CleanShot 2024-09-17 at 12 35 56@2x

File: buildSrc/src/main/kotlin/AndroidBuildGradlePlugin.kt (hunk 2 above)
Comment (witrisna) [reply buildSrc/src/main/kotlin/AndroidBuildGradlePlugin.kt:19]: I have tested it on Android 15, it works fine for now.

File: forgerock-auth/src/main/java/org/forgerock/android/auth/webauthn/WebAuthnRegistration.kt
Comment (witrisna) [reply forgerock-auth/src/main/java/org/forgerock/android/auth/webauthn/WebAuthnRegistration.kt:1]: Updated

File: forgerock-auth/src/test/java/org/forgerock/android/auth/callback/AppIntegrityCallbackTest.kt
Comment (witrisna) [reply forgerock-auth/src/test/java/org/forgerock/android/auth/callback/AppIntegrityCallbackTest.kt:1]: Updated

File: samples/quickstart/src/main/AndroidManifest.xml
Comment (witrisna) [reply samples/quickstart/src/main/AndroidManifest.xml:1]: Updated

**Review by spetrov — APPROVED** [review]

**Pull request merged**